### PR TITLE
[IMP] http_routing: remove trailing / for homepage url prefixed with …

### PR DESCRIPTION
**Pull request by rdeodoo**
…language

### CODE REVIEW

**Comment by rdeodoo** [comment]
To do: add tests redirect, something like
/fr_BE/ -> /fr_BE
/fr_BE/?a=b -> /fr_BE?a=b
CHeck if controller redirect are already tests
/my/ -> /my
/my/?a=b -> /my?a=b

File: addons/http_routing/models/ir_http.py [hunk 1]
```diff
@@ -488,6 +488,11 @@ def _dispatch(cls):
                     return redirect
                 elif url_lg:
                     request.uid = None
+                    if request.httprequest.path == '/%s/' % url_lg:
+                        # special case for homepage controller
```
Comment (rdeodoo) [inline addons/http_routing/models/ir_http.py:492]: more detailed comment + comment it is same as post_process_args + missing if querystring before append

**Comment by rdeodoo** [comment]
- Tests added
- Found and fixed a similar bug for website.page _serve_page trailing slashes
- Removed trailing slash in controller only for frontend routes (removed the diff from backend routes)
- Removed the logger warning when a route had a trailing slash

File: addons/web/controllers/main.py [hunk 2]
```diff
@@ -1423,7 +1423,7 @@ def content_assets(self, id=None, filename=None, unique=None, extra=None, **kw):
     @http.route(['/web/partner_image',
         '/web/partner_image/<int:rec_id>',
         '/web/partner_image/<int:rec_id>/<string:field>',
-        '/web/partner_image/<int:rec_id>/<string:field>/<string:model>/'], type='http', auth="public")
+        '/web/partner_image/<int:rec_id>/<string:field>/<string:model>'], type='http', auth="public")
```
Comment (seb-odoo) [inline addons/web/controllers/main.py:1426]: Hello, just a heads up this route should be removed in https://github.com/odoo/odoo/pull/69819 which is going to be merged soon

**Comment by JKE-be** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**